Code/Framework/AzCore/AzCore/Jobs/Job.h

```diff
@@ -259,7 +270,7 @@ namespace AZ
         {
             countAndFlags |= (unsigned int)FLAG_COMPLETION;
         }
-        countAndFlags |= (unsigned int)((priority << FLAG_PRIORITY_START_BIT) & FLAG_PRIORITY_MASK);
+        countAndFlags |= (unsigned int)((static_cast<uint32_t>(priority) << FLAG_PRIORITY_START_BIT) & FLAG_PRIORITY_MASK);
```

Comment:
Nitpick: Prefer to use `aznumeric_cast<>` or `aznumeric_caster` for numeric casts.

Reply:
Actually why is this? For the std stubs I can understand, but static_cast and reinterpret_cast etc are built-ins

Reply:
`aznumeric_cast<>` and `aznumeric_caster` both check if there isn't any precision loss when casting down. This came from a previous Principle having to fix issues where `static_cast` was used on values that were never suppose to be bigger an expected limit over time growing over that limit. `aznumeric_cast*` checks for these cases in debug and profile builds. The `aznumeric_caster` is just a bit of extra niceness to avoid having to update code when the target type changes as its automatically detected. For cases where precision loss is explicitly expected (such as slicing of pieces of a larger hash) `azlossy_cast<>` and `azlossy_caster` can be used. This better communicates the loss is intentional than `static_cast` can. In release both versions are just plain casts.